src/Test/TestAIMuriel.java

```diff
@@ -112,24 +145,30 @@ private boolean testHuntStall()
 
     // Ask Muriel what to do.
     testCo1.initTurn(testMap);
-    GameAction act = testCo1.getNextAIAction(testMap);
-    performGameAction(act, testGame);
+    GameAction act = null;
+    do
+    {
+      act = testCo1.getNextAIAction(testMap);
+      if( null != act )
+        testPassed &= validate(performGameAction(act, testGame), "    "+ai.getName()+" generated a bad action!");
```

Comment:
Same set of objections to this being a) a loop when it probably isn't needed, and 2) not being "universally true" in that an AI can be working correctly even when not "passing" this test.

Reply:
Added the loop 'cause Wally builds stuff before allocating travel, and order isn't really important there.
Blocking your own factory is *almost always* real dumb, and isn't something I'd ever expect an AI to do intelligently... and to counter a *Recon* slightly better? Nah.